hazelcast/src/main/java/com/hazelcast/scheduledexecutor/impl/ScheduledExecutorContainer.java

```diff
@@ -0,0 +1,426 @@
+/*
+ * Copyright (c) 2008-2016, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.scheduledexecutor.impl;
+
+import com.hazelcast.logging.ILogger;
+import com.hazelcast.scheduledexecutor.DuplicateTaskException;
+import com.hazelcast.scheduledexecutor.ScheduledTaskHandler;
+import com.hazelcast.scheduledexecutor.ScheduledTaskStatistics;
+import com.hazelcast.scheduledexecutor.StaleTaskException;
+import com.hazelcast.scheduledexecutor.StatefulTask;
+import com.hazelcast.scheduledexecutor.impl.operations.ResultReadyNotifyOperation;
+import com.hazelcast.scheduledexecutor.impl.operations.SyncStateOperation;
+import com.hazelcast.spi.InvocationBuilder;
+import com.hazelcast.spi.NodeEngine;
+import com.hazelcast.spi.Operation;
+import com.hazelcast.spi.OperationService;
+import com.hazelcast.spi.impl.executionservice.InternalExecutionService;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.concurrent.Callable;
+import java.util.concurrent.Delayed;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicReference;
+
+import static com.hazelcast.scheduledexecutor.impl.DistributedScheduledExecutorService.SERVICE_NAME;
+import static com.hazelcast.util.ExceptionUtil.sneakyThrow;
+
+public class ScheduledExecutorContainer {
+
+    private final ILogger logger;
+
+    private final String name;
+
+    private final NodeEngine nodeEngine;
+
+    private final InternalExecutionService executionService;
+
+    private final int partitionId;
+
+    private final int durability;
+
+    private final Map<String, ScheduledTaskDescriptor> tasks;
+
+    private final Map<String, BackupTaskDescriptor> backups;
```

Comment:
since we have a `member-partition-container` it means it is accessed from generic threads. and this means we need to have ConcurrentMap over here too.